benchmark-e2e/src/main/java/io/opentelemetry/e2ebenchmark/E2EAgentBenchmark.java

```diff
@@ -0,0 +1,111 @@
+package io.opentelemetry.e2ebenchmark;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.lifecycle.Startables;
+import org.testcontainers.utility.DockerImageName;
+import lombok.extern.log4j.Log4j2;
+import org.testcontainers.utility.MountableFile;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Log4j2
+public class E2EAgentBenchmark {
+  private static final Logger LOG = LoggerFactory.getLogger(E2EAgentBenchmark.class);
+  private List<GenericContainer<?>> containers;
+
+  // configurable values
+  private String APP = System.getenv("APP_IMAGE");
```

Comment:
How about defaults based on one of our smoke test images?

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SpringBootSmokeTest.groovy#L20

In fact, if you can follow the pattern of the smoke tests to build the agent and add it to the container, then the benchmark will always be running against the current HEAD which is awesome.

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/smoke-tests/smoke-tests.gradle#L34

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy#L90